tests/unittest_scoped_nodes.py

```diff
@@ -473,6 +473,10 @@ def test_argnames(self) -> None:
         astroid = builder.parse(code, __name__)
         self.assertEqual(astroid["f"].argnames(), ["a", "b", "c", "args", "kwargs"])
 
+        code_with_kwonly_args = "def f(a, b, *, c=None, d=None): pass"
```

Comment:
If you change `*` into `*args`, that could test the correct order. Maybe add `**kwargs` as well.

Furthermore, maybe a bit outside the scope of the PR, it could make sense to test positional only arguments too.